internal/syscallfs/dirfs.go

```diff
@@ -34,20 +45,16 @@ func (dir dirFS) Path() string {
 
 // OpenFile implements FS.OpenFile
 func (dir dirFS) OpenFile(name string, flag int, perm fs.FileMode) (fs.File, error) {
-	f, err := os.OpenFile(path.Join(string(dir), name), flag, perm)
+	f, err := os.OpenFile(dir.join(name), flag, perm)
 	if err != nil {
 		return nil, err
 	}
-
-	if flag == 0 || flag == os.O_RDONLY {
```

Comment:
instead of masking here, we are verifying the errno on Write. seems to work e.g. flags are honored to prohibit writes.